Content.Shared/SimpleStation14/Silicon/Systems/SharedSiliconChargerSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Content.Shared.Power;
+using Content.Shared.Storage.Components;
+using Robust.Shared.Audio;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.SimpleStation14.Silicon.Charge;
+
+public sealed class SharedSiliconChargerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SiliconChargerComponent, StorageAfterOpenEvent>(HandleStateOpen);
+        SubscribeLocalEvent<SiliconChargerComponent, StorageAfterCloseEvent>(HandleStateClose);
+    }
+
+
+    /// <summary>
+    ///     Updates the state of the charger when it's open or closed.
+    /// </summary>
+    private void HandleStateOpen(EntityUid uid, SiliconChargerComponent component, ref StorageAfterOpenEvent _)
+    {
+        UpdateState(uid, component);
+    }
+
+    /// <inheritdoc cref="HandleStateOpen"/>
+    private void HandleStateClose(EntityUid uid, SiliconChargerComponent component, ref StorageAfterCloseEvent _)
+    {
+        UpdateState(uid, component);
+    }
+
+    /// <summary>
+    ///     Updates the visual and auditory state of the charger based on if it's active, and/or open.
+    /// </summary>
+    public void UpdateState(EntityUid uid, SiliconChargerComponent? component = null)
```

Comment:
@DEATHB4DEFEAT Let me know if you know of anyway to use this the way I need to, without making it Public.